File: rest_api/app/schemas/session.py
"""Pydantic schemas for session join endpoint."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class SessionJoinRequest(BaseModel):
    """Request body for POST /api/sessions/join."""

    branchSlug: str = Field(..., min_length=1, max_length=100)
    tableIdentifier: str = Field(..., min_length=1, max_length=20)
    displayName: str = Field(default="", max_length=100)
    avatarColor: str = Field(default="#f97316", max_length=20)
    locale: str = Field(default="es", max_length=10)


class BranchInfo(BaseModel):
    """Branch summary returned in the session join response."""

    id: int
    name: str
    slug: str


class TableInfo(BaseModel):
    """Table summary returned in the session join response."""

    identifier: str
    displayName: str


class SessionJoinResponse(BaseModel):
    """Response body for POST /api/sessions/join."""

    token: str
    sessionId: int
    expiresAt: datetime
    branch: BranchInfo
    table: TableInfo

File: pwa_menu/src/types/session.ts
/**
 * Session types for the anonymous customer session flow.
 * QR → Landing → POST /api/sessions/join → Token stored in Zustand persist
 */

/** Request body for POST /api/sessions/join */
export interface SessionJoinRequest {
  branchSlug: string;
  tableIdentifier: string;
  displayName: string;
  avatarColor: string;
  locale: string;
}

/** Response body from POST /api/sessions/join */
export interface SessionJoinResponse {
  token: string;
  sessionId: string;
  expiresAt: string; // ISO 8601
  branch: {
    id: string;
    name: string;
    slug: string;
  };
  table: {
    identifier: string;
    displayName: string;
  };
}

/** State shape persisted in localStorage under 'buen-sabor-session' */
export interface SessionState {
  /** HMAC token sent as X-Table-Token header */
  token: string | null;
  /** UUID v4 session identifier */
  sessionId: string | null;
  displayName: string;
  avatarColor: string;
  branchSlug: string | null;
  branchName: string | null;
  tableIdentifier: string | null;
  tableName: string | null;
  /** ISO 8601 timestamp of session creation */
  joinedAt: string | null;
  /** Unix timestamp (ms) of last user activity — sliding window for 8h expiry */
  lastActivity: number | null;
}
